### ISSUE TITLE
GitHub authentication

### ISSUE THREAD

**Issue by StevieDicks** [microsoft/vscode#148055]
Issue Type: <b>Bug</b>

OS: Amazon Linux virtual desktop
Browser: Mozilla Firefox

When trying to authenticate GitHub using the sign-in on browser feature I am unable to authenticate. I get to the "You are being redirected to the authorized application..." page and nothing happens. In the line "If your browser does not redirect you back, please click here to continue..." the 'click here' link does nothing. If I try to sign on by typing my GitHub username and token into the command palette the "signing into GitHub" notification just spins for a minute or so then cancels.

I am signed into GitHub in the terminal, where my PAT for this machine works fine. 

VS Code version: Code 1.66.2 (dfd34e8260c270da74b5c2d86d61aee4b6d56977, 2022-04-11T07:49:24.808Z)
OS version: Linux x64 4.14.275-207.503.amzn2.x86_64
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Xeon(R) Platinum 8259CL CPU @ 2.50GHz (2 x 2499)|
|GPU Status|2d_canvas: unavailable_software<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: disabled_software<br>multiple_raster_threads: disabled_off<br>oop_rasterization: disabled_off<br>opengl: disabled_off<br>rasterization: disabled_software<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software|
|Load (avg)|2, 1, 0|
|Memory (System)|3.79GB (0.48GB free)|
|Process Argv|. --crash-reporter-id f3a5e405-ce50-429e-9dd2-cae07d024289|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|mate|
|XDG_CURRENT_DESKTOP|MATE|
|XDG_SESSION_DESKTOP|undefined|
|XDG_SESSION_TYPE|undefined|
</details><details><summary>Extensions (2)</summary>

Extension|Author (truncated)|Version
---|---|---
go|gol|0.32.0
cpptools|ms-|1.9.8

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30443607
pythontb:30283811
pythonvspyt551:30345470
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
pythonvsnew555:30457759
vscscmwlcmt:30465135
cppdebugcf:30475217

```

</details>

**Comment by StevieDicks** [comment]
> /duplicate [#147104 (comment)](https://github.com/microsoft/vscode/issues/147104#issuecomment-1109108734)

The solution you linked does not solve the issue for me

**Comment by TylerLeonhardt** [comment]
Did you try in Insiders because this hasn't been released to Stable yet (scheduled for early next week)

**Comment by StevieDicks** [comment]
Using Insiders, I followed the instructions in the linked issue. 

Eventually a new prompt came up, asking to authenticate in a different way. I responded "yes", and it gave me a code to submit on GitHub. This worked! Thank you for the help.

Persisting issue (not blocking me -- this is just for your info):
Although I am now logged into GitHub on VS Code, Settings Sync forces authentication a second time. If I do the same method of authentication then it works, but using the default does not work. 

Thanks again.
